### PR TITLE
[HEAP] add tracking server and cdn

### PR DESCRIPTION
**Pull request by murphpdx**
Updated the heap integration to allow users to load heap from their own cdn and set the tracking server. 

**There are configuration changes. That means you'll need to update our customer's config in the database so they do not experience disruption. Thank you.**

## Testing

- [ ] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [ ] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [ ] [Segmenters] Tested in the staging environment

### CODE REVIEW

**Comment by joe-ayoub-segment** [comment]
Hi @murphpdx , thanks for raising this PR. I'll schedule it for review.

File: packages/browser-destinations/src/destinations/heap/generated-types.ts [hunk 1]
```diff
@@ -13,4 +13,12 @@ export interface Settings {
    * This option is turned off by default to accommodate websites not served over HTTPS. If your application uses HTTPS, we recommend enabling secure cookies to prevent Heap cookies from being observed by unauthorized parties. For more information visit the heap [docs page](https://developers.heap.io/docs/web#securecookie).
    */
   secureCookie?: boolean
+  /**
+   * This is an optional setting. This is used to set up first-party data collection. For most cased this should not be set. For more information visit the heap [docs page](https://developers.heap.io/docs/set-up-first-party-data-collection-in-heap).
+   */
+  trackingServer?: string
+  /**
+   * This is an optional setting used to set the host that loads heap-js. This setting is used when heapJS is self-hosted. In most cased this should be left unset. The hostname should not contain https or app id it will be populated like so: https://${hostname}/js/heap-${appId}.js. For more information visit the heap [docs page](https://developers.heap.io/docs/heap-js#hostname).
```
Comment (j0narch3r) [inline packages/browser-destinations/src/destinations/heap/generated-types.ts:21]: I think the docs url should be https://developers.heap.io/docs/self-hosting-heapjs

File: packages/browser-destinations/src/destinations/heap/index.ts [hunk 2]
```diff
@@ -53,6 +54,20 @@ export const destination: BrowserDestinationDefinition<Settings, HeapApi> = {
...
+      label: 'Tracking Server',
+      description:
+        'This is an optional setting. This is used to set up first-party data collection. For most cased this should not be set. For more information visit the heap [docs page](https://developers.heap.io/docs/set-up-first-party-data-collection-in-heap).',
+      type: 'string',
+      required: false
+    },
+    hostname: {
+      label: 'Hostname',
+      description:
+        'This is an optional setting used to set the host that loads heap-js. This setting is used when heapJS is self-hosted. In most cased this should be left unset. The hostname should not contain https or app id it will be populated like so: https://${hostname}/js/heap-${appId}.js. For more information visit the heap [docs page](https://developers.heap.io/docs/heap-js#hostname).',
```
Comment (j0narch3r) [inline packages/browser-destinations/src/destinations/heap/index.ts:68]: I think the docs url should be https://developers.heap.io/docs/self-hosting-heapjs

**Review by j0narch3r — COMMENTED** [review]
Very minor docs corrections. Otherwise looks solid.

**Review by pooyaj — APPROVED** [review]

**Pull request merged**